### ISSUE TITLE
Why does `status` use a callback?

### ISSUE THREAD

**Issue by Danappelxx** [ennioma/arek#33]
When would asking for a permission be asynchronous? All of the permissions in this repository which I looked at are not asynchronous. Seems like an unnecessary evil, doesn't it?

**Comment by ennioma** [comment]
Hi,
I definitely agree with you. Initially I wrote this library to explore swift3 so I missed something in the design.

This is one of the key point I have on my list to update shortly.

Thanks!

**Comment by Danappelxx** [comment]
Great!

**Comment by ennioma** [comment]
Hi @Danappelxx,
I found why I did it :)

Take a look at [ArekNotifications](https://github.com/ennioma/arek/blob/master/arek/Classes/Permissions/ArekNotifications.swift#L29-L50): the methods to ask for permission status are asynchronous and I also think that this is the way Apple will manage the permissions in the future.

So for now I'd prefer to not modify the current implementation.

Let me know what you think about this.
